### PR TITLE
Switch to Yarn v3 as the package manager default

### PR DESCRIPTION
**Pull request by benpsnyder**
Yarn v3 has some nice innovations. I am proposing to use Yarn over NPM as the default for this repository and am sending this PR as an example of what we have implemented as a standard at our company.  We perform `yarn set version berry` then `yarn set version 3.x` because there is a bug in the stable version.

### CODE REVIEW

**Comment by srleecode** [comment]
Related issues in nx which look like they should be fixed before this one can be merged: [nrwl/nx#7504](https://github.com/nrwl/nx/pull/7504) and [nrwl/nx#7918](https://github.com/nrwl/nx/pull/7918).

With the other changes in the recent 13.2.3 release, I couldn't figure out the spawnSync issue and related issues that occur when running the e2e tasks using affected. For now, I will be running all of the e2e jobs in sequence and not using affected for the e2e test runs. 

When running the E2E with this branch, I was getting errors on the package manager install saying that the yarn command failed.

**Comment by benpsnyder** [comment]
@srleecode `affected` is working now that I upgraded to `@nrwl/nx-cloud@next` which is v13

**Comment by srleecode** [comment]
> @srleecode `affected` is working now that I upgraded to `@nrwl/nx-cloud@next` which is v13

I checked out your branch, ran yarn then the affected e2e and I get this error. I get the same error running yarn e2e.

/tmp/nx-e2e --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --packageManager=yarn --preset=empty
    Command failed: touch yarn.lock && yarn

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,38 +22,40 @@ jobs:
...
   pr:
     runs-on: ubuntu-latest
     if: ${{ github.event_name == 'pull_request' }}
     steps:
       - uses: actions/checkout@v2
         with:
-          ref: ${{ github.event.pull_request.head.ref }}
+          # see: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/
+          # regarding `pull_request_target`
+          ref: ${{ github.event.pull_request_target.head.ref }}
```
Comment (benpsnyder) [inline .github/workflows/ci.yml:43]: @srleecode please get this functionality into `master` so PRs from forks can work with the CI

**Comment by benpsnyder** [comment]
> > @srleecode `affected` is working now that I upgraded to `@nrwl/nx-cloud@next` which is v13
> 
> I checked out your branch, ran yarn then the affected e2e and I get this error. I get the same error running yarn e2e.
> 
> /tmp/nx-e2e --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --packageManager=yarn --preset=empty Command failed: touch yarn.lock && yarn

try now... I an using `yarn dlx` now and `touch` from node

**Comment by srleecode** [comment]
> > I checked out your branch, ran yarn then the affected e2e and I get this error. I get the same error running yarn e2e.
> > /tmp/nx-e2e --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --packageManager=yarn --preset=empty Command failed: touch yarn.lock && yarn
> 
> try now... I an using `yarn dlx` now and `touch` from node

Locally it is working. However, on the CI job it failed: https://github.com/srleecode/domain/runs/4353924313?check_suite_focus=true

    Command failed: node /home/runner/work/domain/domain/node_modules/@nrwl/tao/index.js new proj --nx-workspace-root=/home/runner/work/domain/domain/tmp/nx-e2e --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --packageManager=yarn --preset=empty 
    Command failed: yarn dlx touch yarn.lock && yarn

**Comment by srleecode** [comment]
closing as https://github.com/nrwl/nx/pull/7918 was closed

**Pull request closed without merging**